File: examples/backends/fastapi/app/routes/health.py
"""GET /health — Liveness check."""

from fastapi import APIRouter

from .. import cedar_engine

router = APIRouter()


@router.get("/health")
async def health():
    return {
        "status": "ok",
        "mode": "token-handler",
        "cedar": "ready" if cedar_engine.is_initialized() else "unavailable",
    }

File: rust/src/routes/health.rs
//! GET /health

use axum::Json;
use axum::extract::State;
use std::sync::Arc;

use crate::types::HealthResponse;

/// Health check — returns OK + Cedar status.
pub async fn health(State(state): State<Arc<crate::AppState>>) -> Json<HealthResponse> {
    let cedar_status = if state.cedar.is_some() {
        "ready"
    } else {
        "unavailable"
    };
    Json(HealthResponse {
        status: "ok".into(),
        mode: "token-handler".into(),
        cedar: cedar_status.into(),
    })
}
